src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -56,20 +62,34 @@ private BoundExpression RewriteCollectionExpressionConversion(Conversion convers
             }
         }
 
+        private BoundExpression VisitImmutableArrayCollectionExpression(BoundCollectionExpression node, TypeSymbol elementType)
+        {
+            var elementTypeWithAnnotations = TypeWithAnnotations.Create(elementType);
+            var arrayCreation = VisitArrayOrSpanCollectionExpression(
+                node,
+                CollectionExpressionTypeKind.Array,
+                ArrayTypeSymbol.CreateSZArray(_compilation.Assembly, elementTypeWithAnnotations),
+                elementTypeWithAnnotations);
+            var asImmutableArray = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_InteropServices_ImmutableCollectionsMarshal__AsImmutableArray_T)!;
+            return _factory.StaticCall(asImmutableArray.Construct(elementType), ImmutableArray.Create(arrayCreation));
+        }
```

Comment:
Just curious, are we planning to further optimize the empty collection case later on? (with `ImmutableArray<T>.Empty`)

Reply:
I think this would be fine, though I am curious what the jit asm difference ends up being when we `AsImmutableArray(Array.Empty<T>)` versus accessing `ImmutableArray<T>.Empty`.